sycl/include/sycl/detail/common.hpp

```diff
@@ -25,6 +25,18 @@ namespace sycl {
 __SYCL_INLINE_VER_NAMESPACE(_V1) {
 namespace detail {
 
+// The check for output iterator is commented out as it blocks set_final_data
+// with void * argument to be used.
+// TODO: Align these checks with the SYCL specification when the behaviour
+// with void * is clarified.
+template <typename DataT>
+using EnableIfOutputPointerT = detail::enable_if_t<
+    /*is_output_iterator<DataT>::value &&*/ std::is_pointer<DataT>::value>;
+
+template <typename DataT>
+using EnableIfOutputIteratorT = detail::enable_if_t<
+    /*is_output_iterator<DataT>::value &&*/ !std::is_pointer<DataT>::value>;
```

Comment:
Is the commented-out conjunction there because of the TODO or can they be removed?

Reply:
Yes.